### PR TITLE
chore: sort scripts in package.json

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Alphabetize scripts `package.json`

## Context:

I've done this with the `sort ascending` function in VSCode, I've not used a Prettier plugin or anything like that.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,33 +8,33 @@
...
+    "doc-fix": "run-s markdown-lint-fix prettier-fix",
     "eslint-fix": "eslint --ext .js,.mjs,.ts --fix lib/ test/ tools/",
-    "jest": "cross-env NODE_ENV=test LOG_LEVEL=fatal node --expose-gc node_modules/jest/bin/jest.js",
+    "eslint": "eslint --ext .js,.mjs,.ts lib/ test/ tools/",
+    "generate:imports": "node tmp/tools/generate-imports.js",
+    "generate": "run-s generate:*",
     "jest-debug": "cross-env NODE_ENV=test LOG_LEVEL=fatal node --expose-gc --inspect-brk node_modules/jest/bin/jest.js",
     "jest-silent": "cross-env NODE_ENV=test yarn jest --reporters jest-silent-reporter",
-    "lint": "run-s ls-lint eslint prettier markdown-lint",
+    "jest": "cross-env NODE_ENV=test LOG_LEVEL=fatal node --expose-gc node_modules/jest/bin/jest.js",
```
Comment (viceice) [inline package.json:24]: 🤔 Doesn't need a shorter string come before a longer one?

Comment (HonkingGoose) [reply package.json:24]: Do you want the sorting to be like this then instead?

From top to bottom:

1. `jest`
2. `jest-a`
3. `jest-b`

Comment (viceice) [reply package.json:24]: Yes, this is what i understand as `Alphabetize` 🙃

**Comment by viceice** [comment]
`prepare:generate` needs to be before `generate:imports`, otherwise build will fail 😉

**Comment by HonkingGoose** [comment]
> `prepare:generate` needs to be before `generate:imports`, otherwise build will fail 😉

That seems odd to me, why should the changing the order of scripts in the `package.json` cause the build to fail? What am I missing here?

Shouldn't we make it so script order doesn't affect the build process?

**Comment by viceice** [comment]
> > `prepare:generate` needs to be before `generate:imports`, otherwise build will fail 😉
> 
> That seems odd to me, why should the changing the order of scripts in the `package.json` cause the build to fail? What am I missing here?
> 
> Shouldn't we make it so script order doesn't affect the build process?

`prepare:tools` will compile the tools ts files to js ( eg `tools/generate-imports.ts` ),

`"prepare:generate": "run-s generate:*",` call the generate scripts with `npm-run-all` which runs scripts in package.json key order.

`"generate:imports": "node tmp/tools/generate-imports.js"`, fails without the compiled tools.

I think i will create a new pr to convert the `tools/generate-imports.ts` to esm, as most others are already esm too, so we don't need to compile tools. 🙃

**Comment by HonkingGoose** [comment]
> `"prepare:generate": "run-s generate:*",` call the generate scripts with `npm-run-all` which runs scripts in package.json key order.

Ah okay, so that's what I'm breaking by putting the keys in a different order. 😄 

> I think i will create a new pr to convert the `tools/generate-imports.ts` to esm, as most others are already esm too, so we don't need to compile tools. 🙃

Would that fix the "hardcoded script key order" thing? If so I'll label my PR `status:blocked` until your PR gets merged.
There's no point in me working on this PR if we end up with a "sorta-kinda alphabetical ordering with exceptions". 😉

**Comment by HonkingGoose** [comment]
Labeling this `status:blocked` as I need PR #9887 for my PR to work properly. 😉

**Comment by viceice** [comment]
#9887 merged :wink:

**Comment by HonkingGoose** [comment]
I'm no longer breaking the build with my changes! Hooray! 🥳 🎉 ✨

**Review by viceice — APPROVED** [review]

**Pull request merged**